packages/generator/src/generator-cli.ts

```diff
@@ -46,9 +48,15 @@ export function parseCmdArgs(): GeneratorOptions {
     .alias('config', 'c')
     .alias('version', 'v')
     .alias('help', 'h')
-    .middleware(() => {
-      warnIfDeprecated(process.argv);
-    })
     .strict(true)
-    .recommendCommands().argv as unknown as GeneratorOptions;
+    .parserConfiguration({
+      'strip-aliased': true,
+      'strip-dashed': true
+    })
+    .recommendCommands();
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const { _, $0, ...userOptions } = parsedOptions.argv as any;
```

Comment:
[q] is this part of the stable yargs command - test is definatly needed to see if this breaks.

Reply:
The two parameters `_` and `$0` are part of the yargs API and I think this is very stable. However, those two parameters are not used. What I am doing here is basically remove them from the object - if they are not there, they are undefined, but we don't care.

I just investigated a bit: the problem is that `.argv` could be a promise. I would suggest to take care of this in the follow up, when we [improve the API with `GeneratorOptions`](https://github.com/SAP/cloud-sdk-backlog/issues/980). I partially started it here, but don't want to do everything in one PR.